plugins/woocommerce-blocks/tests/e2e/tests/checkout/additional-fields.shopper.block_theme.spec.ts

```diff
@@ -908,5 +908,101 @@ test.describe( 'Shopper → Additional Checkout Fields', () => {
 				'This is a nice gift'
 			);
 		} );
+
+		test( 'Shopper can refresh the page, and not loose any entered data', async ( {
+			checkoutPageObject,
+		} ) => {
+			await checkoutPageObject.editShippingDetails();
+			await checkoutPageObject.unsyncBillingWithShipping();
+			await checkoutPageObject.editBillingDetails();
+
+			await checkoutPageObject.fillInCheckoutWithTestData(
+				{},
+				{
+					contact: {
+						'Enter a gift message to include in the package':
+							'Happy Birthday!',
+						'Is this a personal purchase or a business purchase?':
+							'personal',
+					},
+					order: {
+						'What is your favourite colour?': 'Red',
+						// 'How did you hear about us?': 'Friend',
+					},
+				}
+			);
+
+			await checkoutPageObject.page
+				.getByLabel(
+					'Do you want to subscribe to our newsletter? (optional)'
+				)
+				.check();
+
+			// Check that contact and order fields are persisted
+			await checkoutPageObject.page.waitForRequest(
+				'**/wc/store/v1/checkout**'
+			);
+
+			// Check that address fields are persisted
+			await checkoutPageObject.page.waitForRequest(
+				'**/wc/store/v1/batch**'
+			);
```

Comment:
Is it normal for both to be triggered?

Reply:
We're filling in the address as well, so yes. I thought it'd be good to check that address fields also get persisted.